llvm/include/llvm/CodeGen/MachineTraceMetrics.h

```diff
@@ -102,19 +103,22 @@ class MachineTraceMetrics : public MachineFunctionPass {
   TargetSchedModel SchedModel;
 
 public:
+  friend class MachineTraceMetricsWrapperPass;
   friend class Ensemble;
   friend class Trace;
 
   class Ensemble;
 
-  static char ID;
+  // For legacy pass.
+  MachineTraceMetrics() {
+    std::fill(std::begin(Ensembles), std::end(Ensembles), nullptr);
+  }
```

Comment:
This is odd to split the constructors, do it in init?

Reply:
It isn't possible to move out the Ensembles pointer initialization from the constructors. Certain tests crashed while the destructor invokes clear() that tries to delete the Ensemble pointers (some garbage value). The default constructor for these tests doesn't appropriately clear the object's members. The tests that crashed don't contain any function definitions, but only some global declarations. So the run() instances won't be invoked for clearing these pointers. Also, they should point to the dynamically allocated addresses (otherwise the initial null addresses) before the destructor is invoked.

Reply:
So the Ensembles should just get = 0 initializer in the declaration? Should also probably be using unique_ptr